app/controllers/api/v2/config_reports_controller.rb

```diff
@@ -39,6 +39,12 @@ def show
       param_group :config_report, :as => :create
 
       def create
+        unless Setting[:enable_reports_storage]
+          render_error 'custom_error', :status => :unprocessable_entity,
+                       :locals => { :message => "Reports feature is disabled, you can enable it in Settings > General" }
+          return
+        end
```

Comment:
The idea is not to throw any error when facts/reports storage is disabled, but to return 200 HTTP OK but doing nothing.

Reply:
But @ekohl wanted to return error status code. So how it should be?

Reply:
IMHO as an API user I want to know that the functionality actually doesn't work. A HTTP status 405 Method Not Allowed sounds like the correct thing to do. Then the integration scripts can properly error out instead of pretending that it still works.